src/libs/actions/Policy/Policy.ts

```diff
@@ -3740,6 +3749,634 @@ function getAdminPoliciesConnectedToNetSuite(): Policy[] {
+ */
+function enableAutoApprovalOptions(policyID: string, enabled: boolean) {
+    const autoApprovalValues = !enabled ? {autoApproval: {auditRate: CONST.POLICY.RANDOM_AUDIT_DEFAULT_PERCENTAGE, limit: CONST.POLICY.AUTO_APPROVE_REPORTS_UNDER_DEFAULT_CENTS}} : {};
+    const autoApprovalCleanupValues = !enabled ? {autoApproval: null} : {};
+    const autoApprovalPendingValues = !enabled ? {autoApproval: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE} : {};
+
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                shouldShowAutoApprovalOptions: enabled,
+                ...autoApprovalValues,
+                pendingFields: {
+                    shouldShowAutoApprovalOptions: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+                    ...autoApprovalPendingValues,
+                },
+            },
+        },
+    ];
+
+    const successData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                pendingFields: {
+                    shouldShowAutoApprovalOptions: null,
+                    ...autoApprovalCleanupValues,
+                },
+                errorFields: null,
+            },
+        },
+    ];
+
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                shouldShowAutoApprovalOptions: !enabled,
+                pendingFields: {
+                    shouldShowAutoApprovalOptions: null,
+                    ...autoApprovalCleanupValues,
+                },
+                errorFields: {
+                    shouldShowAutoApprovalOptions: ErrorUtils.getMicroSecondOnyxErrorWithTranslationKey('common.genericErrorMessage'),
+                },
+            },
+        },
+    ];
+
+    const parameters: EnablePolicyAutoApprovalOptionsParams = {
+        enabled,
+        policyID,
+    };
+
+    API.write(WRITE_COMMANDS.ENABLE_POLICY_AUTO_APPROVAL_OPTIONS, parameters, {
+        optimisticData,
+        successData,
+        failureData,
+    });
+}
+
+/**
+ * Call the API to set the limit for auto-payments in the given policy
+ * @param policyID - id of the policy to apply the limit to
+ * @param limit - max amount for auto-payment for the reports in the given policy
+ */
+function setPolicyAutoReimbursementLimit(policyID: string, limit: string) {
+    const policy = getPolicy(policyID);
+    const fallbackLimit = limit === '' ? '0' : limit;
+    const parsedLimit = CurrencyUtils.convertToBackendAmount(parseFloat(fallbackLimit));
+
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.FORMS.RULES_AUTO_PAY_REPORTS_UNDER_MODAL_FORM,
```

Comment:
Why do we need to update this field?

Reply:
I believe it's a leftover from the error state debug from that form. Removed it